### PR TITLE
8225: Adding a checkbox to StringFilterForm to control whether an empty value should cause the filter to be skipped

### LINKED ISSUE #8225: Return all items by projection in ContainsAll if provided value is empty
Need a filter for projections over StringFields that allows returning all items when the provided value is empty.
Like the filter ContainsAnyIfProvided i need a ContainsAllIfProvided filter
I will make a pr to do this.

### PR DESCRIPTION
**Pull request by BenedekFarkas**
Fixes #8225

The changes from #8213 are also removed as that functionality is now covered, as well as that of #8226.

### CODE REVIEW

**Review by sebastienros — APPROVED** [review]

**Comment by BenedekFarkas** [comment]
@MatteoPiovanelli-Laser @LorenzoFrediani-Laser please see the changes here. If you're already running the code in your PRs linked above, then you can update your Queries with a recipe for example - adapting the filter configurations is a simple search and replace.
Basically, if you're using `ContainsAnyIfProvided` or `ContainsAllIfProvided`, then you should change it to use `ContainsAny` or `ContainsAll` respectively and have the checkbox enabled.

**Comment by AndreaPiovanelli** [comment]
> @MatteoPiovanelli-Laser @LorenzoFrediani-Laser please see the changes here. If you're already running the code in your PRs linked above, then you can update your Queries with a recipe for example - adapting the filter configurations is a simple search and replace. Basically, if you're using `ContainsAnyIfProvided` or `ContainsAllIfProvided`, then you should change it to use `ContainsAny` or `ContainsAll` respectively and have the checkbox enabled.

I believe the safest way to ensure compatibility of the new StringFilterForm operators and to properly add the new boolean parameter where needed inside the FilterRecords is to add a migration step to Orchard.Projections, perhaps using Orchard.Forms.Services.FormParametersHelper to correct the form state.

**Comment by BenedekFarkas** [comment]
> I believe the safest way to ensure compatibility of the new StringFilterForm operators and to properly add the new boolean parameter where needed inside the FilterRecords is to add a migration step to Orchard.Projections, perhaps using Orchard.Forms.Services.FormParametersHelper to correct the form state.

I didn't find a proper way to manipulate the form, but we can still adjust the filter in any way to cover the earlier cases.
BTW are you already using `ContainsAnyIfProvided` in production? Honestly I'm not really keen on adding a migration step just for this, especially on 1.10.x after the yak shaving I had to do with merging 1.10.x into dev the last time due to conflicting migrations.

Also, this is still unreleased code, so I'm not that worried about breaking changes. But if you're already running this in production and you can't update filters easily with recipes, then I'm happy to work towards a solution that works for you too.

File: src/Orchard.Web/Modules/Orchard.Projections/FilterEditors/Forms/StringFilterForm.cs [hunk 1]
```diff
@@ -36,8 +36,14 @@ public class StringFilterForm : IFormProvider {
                             Title: T("Value"),
                             Classes: new[] { "text medium", "tokenized" },
                             Description: T("Enter the value the string should be.")
-                            )
-                        );
+                        ),
+                        _IgnoreIfEmptyValue: Shape.Checkbox(
+                            Id: "IgnoreFilterIfValueIsEmpty",
+                            Name: "IgnoreFilterIfValueIsEmpty",
+                            Title: "Ignore filter if value is empty",
```
Comment (AndreaPiovanelli) [inline src/Orchard.Web/Modules/Orchard.Projections/FilterEditors/Forms/StringFilterForm.cs]: Can we add T("Ignore filter if value is empty"), please?

**Comment by AndreaPiovanelli** [comment]
> > I believe the safest way to ensure compatibility of the new StringFilterForm operators and to properly add the new boolean parameter where needed inside the FilterRecords is to add a migration step to Orchard.Projections, perhaps using Orchard.Forms.Services.FormParametersHelper to correct the form state.
> 
> I didn't find a proper way to manipulate the form, but we can still adjust the filter in any way to cover the earlier cases. BTW are you already using `ContainsAnyIfProvided` in production? Honestly I'm not really keen on adding a migration step just for this, especially on 1.10.x after the yak shaving I had to do with merging 1.10.x into dev the last time due to conflicting migrations.
> 
> Also, this is still unreleased code, so I'm not that worried about breaking changes. But if you're already running this in production and you can't update filters easily with recipes, then I'm happy to work towards a solution that works for you too.

I believe we may use a Repository<FilterRecord>, and we want to edit the column "State", which is similar to: 
`<Form><Description></Description><Operator>ContainsAllIfProvided</Operator><Value>{Request.QueryString:q}</Value></Form>`
These rows need to be updated, with the help of FormParametersHelper, adding the parameter "IgnoreIfEmptyValue" when needed.
The migration ensures every row is updated properly and in a multi-tenant, multi-instance with tens of tenants, using a recipe for each tenant is tricky.

If needed, I can implement the migration phase and give you a branch to cherry pick from or a gist.

File: src/Orchard.Web/Modules/Orchard.Projections/FilterEditors/Forms/StringFilterForm.cs (hunk 1 above)
Comment (BenedekFarkas) [reply src/Orchard.Web/Modules/Orchard.Projections/FilterEditors/Forms/StringFilterForm.cs]: Absolutely, good catch!

Comment (BenedekFarkas) [reply src/Orchard.Web/Modules/Orchard.Projections/FilterEditors/Forms/StringFilterForm.cs]: Fixed in 9481a66fbf1ceef908e7161b4e2ced7a76800787.

**Comment by BenedekFarkas** [comment]
That's all right and I knew how to migrate the data, I just wanted to avoid having it. :)
Please check the latest changes and let me know if it's working OK for you too!

**Comment by AndreaPiovanelli** [comment]
> That's all right and I knew how to migrate the data, I just wanted to avoid having it. :) Please check the latest changes and let me know if it's working OK for you too!

I confirm migration works and query execution seems good.

**Comment by BenedekFarkas** [comment]
> I confirm migration works and query execution seems good.

Awesome, thanks!

**Comment by BenedekFarkas** [comment]
This change is now in `dev` too!

**Pull request merged**